```{r load}
library(data.table)
library(matrixStats)
library(stats)
library(gplots)
library(DESeq2)
library(ggplot2)
library(RColorBrewer)
library(pheatmap)

```

### Read HTSeq data:

```{r Read HTSeq data}
cs15_fore <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs15_fore_htseq_results.txt", head=FALSE)
cs15_hind <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs15_hind_htseq_results.txt", head=FALSE)
cs16_fore <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs16_fore_htseq_results.txt", head=FALSE)
cs16_hind <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs16_hind_htseq_results.txt", head=FALSE)
cs17_fore <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs17_fore_htseq_results.txt", head=FALSE)
cs17_hind <- read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/9_htseq_count/cs17_hind_htseq_results.txt", head=FALSE)
```

### Rename columns
```{r Rename columns}
names(cs15_fore)[2] <- "FL_15"
names(cs15_hind)[2] <- "HL_15"
names(cs16_fore)[2] <- "FL_16"
names(cs16_hind)[2] <- "HL_16"
names(cs17_fore)[2] <- "FL_17"
names(cs17_hind)[2] <- "HL_17"
```

### Merge the data frames
```{r}
countData <- merge(cs15_fore,cs15_hind,by="V1")
countData <- merge(countData,cs16_fore,by="V1")
countData <- merge(countData,cs16_hind,by="V1")
countData <- merge(countData,cs17_fore,by="V1")
countData <- merge(countData,cs17_hind,by="V1")
```

### Rename the first columns
```{r}
names(countData)[1] <- "Genes"
```

### Remove extra information
```{r}
countData <- countData[-c(1,2,3,4,5),]
countData <- data.frame(row.names =countData[,1], FLCS15=countData[2], FLCS16=countData[3], FLCS17=countData[4], HLCS15=countData[5], HLCS16=countData[6], HLCS17=countData[7])
```

### to change the gene names
```{r}
genes_index <- read.csv2("/Users/Carolina/Documents/GitHub/GenomeAnalysis/code/Genes_index.csv", header = TRUE, sep=",")

# check if the names are the same (in the same order)
print(genes_index$name == row.names(countData))

# add column with gene names
countData <- cbind(countData,genes_index$blast_name)
```
### Data for DESeq2 analysis
```{r}
count_for_deseq2 <- countData
rownames(count_for_deseq2) = make.names(count_for_deseq2$`genes_index$blast_name`, unique = TRUE)
count_for_deseq2=count_for_deseq2[c("FL_15","HL_15","FL_16","HL_16","FL_17","HL_17")]
deseq2.matr <- as.matrix(count_for_deseq2)
```

### remove rows where all the counts are zero
```{r}
countData <- countData[apply(countData[c(1,4)], 1, function(x) !all(x==0)),]
```

### change row names
```{r}
rownames(countData) = make.names(countData$`genes_index$blast_name`, unique = TRUE)
```

### remove the extra column with row names
```{r}
countData=countData[c("FL_15","HL_15","FL_16","HL_16","FL_17","HL_17")]
```

### transform into matrix
```{r}
countData.matr <- as.matrix(countData)
```

```{r}
zScoreMatrix <- (countData.matr-rowMeans(countData.matr))/(rowSds(as.matrix(countData.matr)))[row(countData.matr)]
```

```{r}
#row.clus <-hclust(dist(zScoreMatrix))
row.clus <-hclust(dist(countData))
scaleBWR <- colorRampPalette(c("blue","white","red"), space="rgb")(10)
```

### Plotting heatmap
```{r}
par(mar=c(1,1,1,1))

#without top dendogram
heatmap.2(zScoreMatrix, Rowv = as.dendrogram(row.clus),Colv = NA, col=scaleBWR,  density.info = "none", trace = "none", margins = c(7,7))

# with top dendogram
#heatmap.2(zScoreMatrix, Rowv = as.dendrogram(row.clus), col=scaleBWR,  density.info = "none", trace = "none", margins = c(7,10))
```
The z-score scale represents mean-subtracted regularized log-transformed read counts.


# DESeq2

# Compare Hindlimb vs Forelimb
### Load data
```{r}
cts = deseq2.matr
coldata = read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/metadata_limbs.csv", sep=",", header = TRUE)
```

### create dataframe
```{r}
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = ~ condition)
```

### Analyse data
```{r}
dds <- DESeq(dds)

# results extracts a result table from a DESeq analysis giving base means across samples, log2 fold changes, standard errors, test statistics, p-values and adjusted p-values
res <- results(dds, contrast = c("condition", "Forelimb", "Hindlimb"))

#resultsNames returns the names of the estimated effects (coefficents) of the model
resultsNames(dds)
```

```{r}
# Adds shrunken log2 fold changes (LFC) and SE to a results table from DESeq run without LFC shrinkage
reslfc = lfcShrink(dds, coef = "condition_Hindlimb_vs_Forelimb", type = "apeglm")
```

```{r}
resOrdered <- res[order(res$pvalue),]
```
Results ordered by P-value

### Write to file
```{r}
write.csv(as.data.frame(resOrdered), file="/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/HL_FL_deseq2_results.csv")
```

### Write subset with adjusted p-val < 0.01
```{r}
sub_res <- subset(resOrdered, padj <0.01)
write.csv(as.data.frame(sub_res), file="/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/HL_FL_deseq2_results_pval0.01.csv")
```

```{r}
summary(res)
```

### Plot
```{r}
par(mfrow = c(1,1))
```

```{r}
plotMA(res)
plotMA(sub_res)
plotMA(res_dev)
```
Scatter plot of log2 fold changes (on the y-axis) versus the mean of normalized counts (on the x-axis)

Points will be colored red if the adjusted p value is less than 0.1

In DESeq2, the function plotMA shows the log2 fold changes attributable to a given variable over the mean of normalized counts for all the samples in the DESeqDataSet. Points will be colored red if the adjusted p value is less than 0.1. Points which fall out of the window are plotted as open triangles pointing either up or down.


```{r}
plotMA(reslfc)
```
It is more useful visualize the MA-plot for the shrunken log2 fold changes,
which remove the noise associated with log2 fold changes from low count genes without requiring arbitrary filtering thresholds.

###  recover the gene identifiers by saving the resulting indices:
```{r}
## Do this on the console
# plotMA(reslfc)
# idx <- identify(res$baseMean, res$log2FoldChange)
# rownames(res)[idx]
```


```{r}
gene=which.min(res$padj)
p <- plotCounts(dds, gene, intgroup="condition", returnData = TRUE)

ggplot(p, aes(x=condition, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  scale_y_log10(breaks=c(100, 1000, 5000, 10000, 15000)) +
  ggtitle(rownames(res)[gene])+
  theme(plot.title = element_text(hjust = 0.5))

```
Examine the counts of reads for a single gene across the groups
The gene which had the smallest p value from the results table created above # TBX5

### Plot PCA
```{r}
vsd = varianceStabilizingTransformation(dds)
```
This function calculates a variance stabilizing transformation (VST) from the fitted dispersion-mean relation(s) and then transforms the count data (normalized by division by the size factors or normalization factors), yielding a matrix of values which are now approximately homoskedastic (having constant variance along the range of mean values)

```{r}
plotPCA(vsd, intgroup=c("condition", "type"))
```



# Compare Development Stages and limb

### Load data
```{r}
cts = deseq2.matr
coldata_dev = read.table("/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/metadata_dev.csv")
```

### create dataframe
```{r}
dds_dev <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata_dev,
                              design = ~ condition + type)
```

### Analyse data
```{r}
dds_dev <- DESeq(dds_dev)
```

```{r}
# results extracts a result table from a DESeq analysis giving base means across samples, log2 fold changes, standard errors, test statistics, p-values and adjusted p-values
res_dev <- results(dds_dev)

#resultsNames returns the names of the estimated effects (coefficents) of the model
resultsNames(dds_dev)
```

```{r}
# Adds shrunken log2 fold changes (LFC) and SE to a results table from DESeq run without LFC shrinkage
reslfc_dev_16_15 = lfcShrink(dds_dev, coef = "condition_CS16_vs_CS15", type = "apeglm")
reslfc_dev_17_15 = lfcShrink(dds_dev, coef = "condition_CS17_vs_CS15", type = "apeglm")
reslfc_dev_HL_FL = lfcShrink(dds_dev, coef = "type_HL_vs_FL", type = "apeglm")

```

```{r}
resOrdered_dev <- res_dev[order(res_dev$pvalue),]
```
Results ordered by P-value

### Write to file
```{r}
write.csv(as.data.frame(resOrdered_dev), file="/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/dev_deseq2_results.csv")
```

```{r}
summary(res_dev)
```

```{r}
sub_res_dev <- subset(resOrdered_dev, padj <0.01)

write.csv(as.data.frame(sub_res_dev), file="/Users/Carolina/Documents/GitHub/GenomeAnalysis/results/10_deseq2/dev_deseq2_results_pval0.01.csv")
```

### Plot
```{r}
par(mfrow = c(1,1))
```

```{r}
plotMA(res_dev)
plotMA(sub_res_dev)
```
Scatter plot of log2 fold changes (on the y-axis) versus the mean of normalized counts (on the x-axis)

```{r}
plotMA(reslfc_dev_16_15)
plotMA(reslfc_dev_17_15)
plotMA(reslfc_dev_HL_FL)
```

Examine the counts of reads for a single gene across the groups
The gene which had the smallest p value from the results table created above

```{r}
gene2=which.min(res_dev$padj)
p<- plotCounts(dds_dev, gene2, intgroup="condition", returnData = TRUE)

ggplot(p, aes(x=condition, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  scale_y_log10(breaks=c(100, 1000, 5000))+
  ggtitle(rownames(res_dev)[gene2])+
  theme(plot.title = element_text(hjust = 0.5))

ggplot(p, aes(x=condition, y=count)) + 
  geom_point(position=position_jitter(w=0.1,h=0)) + 
  scale_y_log10(breaks=c(100, 1000, 5000))+
  ggtitle(rownames(res)[gene2])+
  theme(plot.title = element_text(hjust = 0.5))

```

### Plot PCA
```{r}
vsd_dev = varianceStabilizingTransformation(dds_dev)
```
This function calculates a variance stabilizing transformation (VST) from the fitted dispersion-mean relation(s) and then transforms the count data (normalized by division by the size factors or normalization factors), yielding a matrix of values which are now approximately homoskedastic (having constant variance along the range of mean values)

```{r}
#plotPCA(vsd_dev, intgroup=c("condition", "type"))
pcaData <- plotPCA(vsd_dev, intgroup=c("condition", "type"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))

ggplot(pcaData, aes(PC1, PC2, color=condition, shape=type)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()

```

```{r}
select <- order(rowMeans(counts(dds_dev,normalized=TRUE)),
                decreasing=TRUE)[1:30]
df <- as.data.frame(colData(dds_dev)[,c("condition","type")])
pheatmap(assay(vsd_dev)[select,], cluster_rows=TRUE, show_rownames=TRUE,
         cluster_cols=FALSE, annotation_col=df)
clustering_distance_rows = "correlation"
```

```{r}
zScoreMatrix <- (cts-rowMeans(cts))/(rowSds(as.matrix(cts)))[row(cts)]

#remove NaN (what were zeros before)
zScoreMatrix <- na.omit(zScoreMatrix)
pheatmap(zScoreMatrix, cluster_rows=TRUE, show_rownames=TRUE,
         cluster_cols=FALSE, fontsize_row = 5)
# can add  annotation_col=df
```

```{r}
genes_pval <- rownames(sub_res_dev)

htseq_genes_pval0.01 <- cts[is.element(rownames(cts), genes_pval),]
htseq_genes_pval0.01

```

```{r}
zScoreMatrix_pval <- (htseq_genes_pval0.01-rowMeans(htseq_genes_pval0.01))/(rowSds(as.matrix(htseq_genes_pval0.01)))[row(htseq_genes_pval0.01)]

#remove NaN (what were zeros before)
zScoreMatrix_pval <- na.omit(zScoreMatrix_pval)
pheatmap(zScoreMatrix_pval, cluster_rows=TRUE, show_rownames=TRUE,
         cluster_cols=FALSE)
```


```{r}
sampleDists <- dist(t(assay(vsd_dev)))

sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd_dev$condition, vsd_dev$type, sep="-")
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors)
```
A heatmap of this distance matrix gives us an overview over similarities and dissimilarities between samples



